objects/client.c

```diff
@@ -3046,6 +3092,15 @@ luaA_client_get_icon_name(lua_State *L, client_t *c)
     return 1;
 }
 
+static int
+luaA_client_set_startup_id(lua_State *L, client_t *c)
+{
+    const char *startup_id = luaL_checkstring(L, -1);
+    c->startup_id = a_strdup(startup_id);
+    luaA_object_emit_signal(L, -3, "property::startup_id", 0);
+    return 0;
+}
```

Comment:
Please use `DO_CLIENT_SET_STRING_PROPERTY(startup_id)` instead of hand-writing this function. That fixes the memory leak of not freeing the old `startup_id` in here, it only emits the signal if the value actually changed and it needs so much less lines of code. Look at the handling of `name`, `role`, `machine` etc.

Reply:
Well, okay, that macro only generates a function for setting the string property. A short function similar to `luaA_client_set_name` is still needed. Still, you get the idea.

Reply:
done